services/compute/virtualmachinescaleset/virtualmachinescaleset.go

```diff
@@ -403,10 +432,48 @@ func (c *client) getWssdVirtualMachineScaleSetHardwareConfiguration(vmp *compute
 				MemoryMB: *vmp.HardwareProfile.CustomSize.MemoryMB,
 			}
 		}
+		if vmp.HardwareProfile.VirtualMachineGPUs != nil {
+			for _, gpu := range vmp.HardwareProfile.VirtualMachineGPUs {
+				if gpu == nil {
+					continue
+				}
+				if gpu.Assignment == nil {
+					return nil, errors.Wrapf(errors.InvalidInput, "GPU assignment is not specified")
+				}
+				var assignment wssdcommon.AssignmentType
+				switch *gpu.Assignment {
+				case compute.GpuDDA:
+					assignment = wssdcommon.AssignmentType_GpuDDA
+				case compute.GpuP:
+					assignment = wssdcommon.AssignmentType_GpuP
+				case compute.GpuPV:
+					assignment = wssdcommon.AssignmentType_GpuPV
+				case compute.GpuDefault:
+					assignment = wssdcommon.AssignmentType_GpuDefault
+				default:
+					return nil, errors.Wrapf(errors.InvalidInput, "Unsupported GPU assignment type [%s]", *gpu.Assignment)
+				}
+				if gpu.PartitionSizeMB == nil {
+					// if partition size is not specified, set it to 0
+					*gpu.PartitionSizeMB = 0
```

Comment:
Woundnt this crash? if the pointer is nil? dont we not need to initialize?

Reply:
I don't think it will crash. We are setting the PartitionSizeMB which is defined as uint64 type, and 0 is a valid value for this data type.